### ISSUE TITLE
The C compiler "/usr/bin/cc" is not able to compile a simple test program.

### ISSUE THREAD

**Issue by gkdr** [dino/dino#4]
Without Ninja, `./configure` just froze.
I installed it, but I get the following error:
```
riba@4820TG ~/dino> ./configure 
-- Found Ninja: /usr/sbin/ninja
-- The C compiler identification is GNU 4.9.4
-- The CXX compiler identification is GNU 4.9.4
-- Check for working C compiler using: Ninja
-- Check for working C compiler using: Ninja -- broken
CMake Error at /usr/share/cmake-2.8/Modules/CMakeTestCCompiler.cmake:61 (message):
  The C compiler "/usr/bin/cc" is not able to compile a simple test program.
  It fails with the following output:
   Change Dir: /home/riba/dino/build/CMakeFiles/CMakeTmp
  Run Build Command:/usr/sbin/ninja cmTryCompileExec4064546234
  die: error: unable to read configuration filelog: reading configuration
  file: cmTryCompileExec4064546234
  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
-- Configuring incomplete, errors occurred!
See also "/home/riba/dino/build/CMakeFiles/CMakeOutput.log".
See also "/home/riba/dino/build/CMakeFiles/CMakeError.log".
```

_CMakeOutput.log_ is:
```
The system is: Linux - 3.13.0-110-generic - x86_64
Compiling the C compiler identification source file "CMakeCCompilerId.c" succeeded.
Compiler: /usr/bin/cc 
Build flags: 
Id flags: 
The output was:
0
Compilation of the C compiler identification source "CMakeCCompilerId.c" produced "a.out"
The C compiler identification is GNU, found in "/home/riba/dino/build/CMakeFiles/2.8.12.2/CompilerIdC/a.out"

Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" succeeded.
Compiler: /usr/bin/c++ 
Build flags: 
Id flags: 
The output was:
0
Compilation of the CXX compiler identification source "CMakeCXXCompilerId.cpp" produced "a.out"
The CXX compiler identification is GNU, found in "/home/riba/dino/build/CMakeFiles/2.8.12.2/CompilerIdCXX/a.out"
```

_CMakeError.log_ is:
```
Determining if the C compiler works failed with the following output:
Change Dir: /home/riba/dino/build/CMakeFiles/CMakeTmp

Run Build Command:/usr/sbin/ninja cmTryCompileExec4064546234
die: error: unable to read configuration filelog: reading configuration file: cmTryCompileExec4064546234
```

Maybe you can help me figure out what's wrong, I'd like to try your client.
If you need any more information, I'd be glad to provide it.

**Comment by mar-v-in** [comment]
I guess you are using a debian or ubuntu system. Try installing the `ninja-build` package via `apt`.

The current detection of existence of ninja-build should be more robust. Currently `./configure` wrongly detects your `/usr/sbin/ninja` from the `ninja` package as ninja-build (which it isn't) and thus is not able to start compilation.

**Comment by gkdr** [comment]
Yes, I am on Ubuntu.
As I said the installation process just froze when I did not have 'ninja' installed, which it recommended. So I just did that.

I pulled the new version and it correctly detects that my ninja is not usable, and continues without freezing. So that issue is fixed too.

Unfortunately I can't continue because the available CMake for my system is too old, and I'm too lazy to fix that manually right now. I assume any other build problems would not be related to this problem anyway, so I'll just close this issue.

Thanks for the help!
